en/lessons/libraries/bypass.md

```diff
@@ -0,0 +1,274 @@
+---
+version: 1.0.0
+title: Bypass
+redirect_from:
+  - /lessons/libraries/bypass/
+---
+
+When testing our applications there are often times we need to make requests to external services, we may even want to simulate different situations like unexpected server errors.
+Handling this in an efficient way doesn't come easy in Elixir without a little help.  
+
+In this lesson we're going to explore how [bypass](https://github.com/PSPDFKit-labs/bypass) can help us quickly and easily handle these requests in our tests.
+
+{% include toc.html %}
+
+## What is Bypass?
+
+[Bypass](https://github.com/PSPDFKit-labs/bypass) is described as "a quick way to create a custom plug that can be put in place instead of an actual HTTP server to return prebaked responses to client requests."
+
+What does that mean?
+Under-the-hood Bypass is an OTP application that masquerades as an external server listening for and responding to requests.
+By responding with pre-defined responses you can test any number of possibilities such as services being unavailable, unexpected server errors, and any expected errors you might encounter.
+
+## Using Bypass
+
+To better illustrate the features of Bypass we'll be building a simple utility application to ping a list of domains and ensure they're online.
+To do this we'll create new supervisor project and a GenServer to check the domains on a configurable interval.
+By leveraging Bypass in our tests we'll be able to verify our application will work in many different outcomes.
+
+_Note_: If you wish to skip ahead to the final code, head over to the Elixir School repo [Clinic](https://github.com/elixirschool/clinic) and have a look.
+
+By this point we should be comfortable creating new Mix projects and adding our dependencies so we'll focus instead of the pieces of code we'll be testing.
```

Comment:
Maybe link [here](https://elixirschool.com/en/lessons/basics/mix/#new-projects) in case someone stumbles upon this but isnt familiar.

Reply:
Great idea!  I'll make this change 👍